source/dem/force_chains_visualization.cc

```diff
@@ -0,0 +1,511 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2020 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ */
+
+#include <core/parameters_lagrangian.h>
+#include <core/tensors_and_points_dimension_manipulation.h>
+
+#include <dem/force_chains_visualization.h>
+#include <dem/particle_particle_contact_force.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+using namespace DEM;
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  ParticlesForceChains(const DEMSolverParameters<dim> &dem_parameters_in)
+  : ParticleParticleContactForce<dim, contact_model, rolling_friction_model>(
+      dem_parameters_in)
+  , dem_parameters(dem_parameters_in)
+{
+  force_normal.push_back(0);
+  vertices.push_back(Point<3>(0, 0, 0));
+  vertices.push_back(Point<3>(0, 0, 0));
```

Comment:
it's allright to initialize to make sure it's never empty, but is this 100% sure necessary?